src/neuroconv/tools/spikeinterface/spikeinterface.py

```diff
@@ -1169,24 +1184,23 @@ def add_units_table(
             data_to_add[property].update(table=nwbfile.electrodes)
 
     # Unit name logic
-    units_ids = checked_sorting.get_unit_ids()
     if "unit_name" in data_to_add:
         unit_name_array = data_to_add["unit_name"]["data"]
     else:
-        unit_name_array = units_ids.astype("str", copy=False)
+        unit_name_array = unit_ids.astype("str", copy=False)
         data_to_add["unit_name"].update(description="Unique reference for each unit.", data=unit_name_array)
 
     # If the channel ids are integer keep the old behavior of asigning table's id equal to unit_ids
-    if np.issubdtype(units_ids.dtype, np.integer):
-        data_to_add["id"].update(data=units_ids.astype("int"))
+    if np.issubdtype(unit_ids.dtype, np.integer):
+        data_to_add["id"].update(data=unit_ids.astype("int"))
 
     units_table_previous_properties = set(units_table.colnames) - set({"spike_times"})
     extracted_properties = set(data_to_add)
     properties_to_add_by_rows = units_table_previous_properties | set({"id"})
     properties_to_add_by_columns = extracted_properties - properties_to_add_by_rows
 
     # Find default values for properties / columns already in the table
-    type_to_default_value = {list: [], np.ndarray: np.array(np.nan), str: "", Real: np.nan}
+    type_to_default_value = {list: [], np.ndarray: np.array(np.nan), str: "", Real: np.nan, bool: False}
```

Comment:
See discussion above https://github.com/catalystneuro/neuroconv/pull/217/files#r1041188364

We used to support this a while back but it causes issues with append functionality, and the last thing we want to do is add incorrect data in those cases.

Please upcast `bool` types to `floats`, however inefficient that might seem, so that we can properly refer to missing values as `NaN`

@h-mayorquin Do you recall if you added any tests to the suite using boolean property types?

Reply:
No, I don't think so. I was thinking on reviewing this on #185 .

Reply:
I see. Maybe casting to string would be better then? At least you still have 'True' and 'False'? With float you would completely lose the categorical aspect

Reply:
I'd be fine with strings, sure